SPECS/picosat/picosat.spec

```diff
@@ -0,0 +1,241 @@
+    -e "s|-Xlinker libpicosat.so|-Xlinker libpicosat.so.0 $RPM_LD_FLAGS -L/usr/lib64/R/lib |" \
+    -e "s/libpicosat/libpicosat-R/g" \
+    -e "s/-lpicosat/-lpicosat-R/g" \
+    -e "s/@TARGETS@/libpicosat-R.so/" \
+  makefile.in > makefile
+%make_build
+
+# Build the version with trace support
+sed -e "s/@CC@/gcc/" \
+    -e "s|@CFLAGS@|$RPM_OPT_FLAGS -D_GNU_SOURCE=1 -DNDEBUG -DTRACE|" \
+    -e "s|-Xlinker libpicosat.so|-Xlinker libpicosat.so.0 $RPM_LD_FLAGS|" \
+    -e "s/libpicosat/libpicosat-trace/g" \
+    -e "s/-lpicosat/-lpicosat-trace/g" \
+    -e "s/@TARGETS@/libpicosat-trace.so picosat picomus/" \
+  makefile.in > makefile
+%make_build
+mv picosat picosat.trace
+
+# Build the fast version.
+# Note that picomus needs trace support, so we don't rebuild it.
+rm -f *.o *.s config.h
+sed -e "s/@CC@/gcc/" \
+    -e "s|@CFLAGS@|$RPM_OPT_FLAGS -D_GNU_SOURCE=1 -DNDEBUG|" \
+    -e "s|-Xlinker libpicosat.so|-Xlinker libpicosat.so.0 $RPM_LD_FLAGS|" \
+    -e "s/@TARGETS@/libpicosat.so picosat picomcs picogcnf/" \
+  makefile.in > makefile
+%make_build
+
+%install
+# Install the header file
+mkdir -p $RPM_BUILD_ROOT%{_includedir}
+cp -p picosat.h $RPM_BUILD_ROOT%{_includedir}
+
+# Install the libraries
+mkdir -p $RPM_BUILD_ROOT%{_libdir}
+install -m 0755 -p libpicosat-R.so \
+  $RPM_BUILD_ROOT%{_libdir}/libpicosat-R.so.0.0.%{version}
+ln -s libpicosat-R.so.0.0.%{version} $RPM_BUILD_ROOT%{_libdir}/libpicosat-R.so.0
+ln -s libpicosat-R.so.0 $RPM_BUILD_ROOT%{_libdir}/libpicosat-R.so
+install -m 0755 -p libpicosat-trace.so \
+  $RPM_BUILD_ROOT%{_libdir}/libpicosat-trace.so.0.0.%{version}
+ln -s libpicosat-trace.so.0.0.%{version} $RPM_BUILD_ROOT%{_libdir}/libpicosat-trace.so.0
+ln -s libpicosat-trace.so.0 $RPM_BUILD_ROOT%{_libdir}/libpicosat-trace.so
+install -m 0755 -p libpicosat.so \
+  $RPM_BUILD_ROOT%{_libdir}/libpicosat.so.0.0.%{version}
+ln -s libpicosat.so.0.0.%{version} $RPM_BUILD_ROOT%{_libdir}/libpicosat.so.0
+ln -s libpicosat.so.0 $RPM_BUILD_ROOT%{_libdir}/libpicosat.so
+
+# Install the binaries
+mkdir -p $RPM_BUILD_ROOT%{_bindir}
+install -m 0755 -p picosat picosat.trace picomus picomcs picogcnf \
+  $RPM_BUILD_ROOT%{_bindir}
+
+# Install the man pages
+mkdir -p $RPM_BUILD_ROOT%{_mandir}/man1
+cp -p %{SOURCE1} %{SOURCE2} %{SOURCE3} $RPM_BUILD_ROOT%{_mandir}/man1
+
+%ldconfig_scriptlets R
+%ldconfig_scriptlets libs
+
+%files
+%{_bindir}/pico*
+%{_mandir}/man1/picosat*
+%{_mandir}/man1/picomus*
+
+%files R
+%doc NEWS
+%{!?_licensedir:%global license %%doc}
+%license LICENSE
+%{_libdir}/libpicosat-R.so.*
+
+%files libs
+%doc NEWS
+%{!?_licensedir:%global license %%doc}
+%license LICENSE
+%{_libdir}/libpicosat-trace.so.*
+%{_libdir}/libpicosat.so.*
+
+%files devel
+%{_includedir}/picosat.h
+%{_libdir}/libpicosat-R.so
+%{_libdir}/libpicosat-trace.so
+%{_libdir}/libpicosat.so
+
+%changelog
+* Mon Jun 21 2021 Rachel Menge <rachelmenge@microsoft.com> 965-13
```

Comment:
>  [](http://example.com/codeflow?start=58&length=1)

Nit, dash here to match our formatting would be neat

Reply:
done:)